tooling/util/proxies.js

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) 2015-2018 Cisco Systems, Inc. See LICENSE file.
+ */
+
+const crypto = require('crypto');
+const http = require('http');
+const path = require('path');
+const url = require('url');
+const yakbak = require('yakbak');
+
+// NOTE: CONVERSATION_SERVICE does not end in URL
+const services = [
+  {
+    defaultUrl: 'https://atlas-a.wbx2.com/admin/api/v1',
+    env: 'ATLAS_SERVICE_URL',
+    name: 'atlas',
+    port: 3010,
+    serviceUrl: process.env.ATLAS_SERVICE_URL
+  },
+  {
+    defaultUrl: 'https://conv-a.wbx2.com/conversation/api/v1',
+    env: 'CONVERSATION_SERVICE',
+    name: 'conversation',
+    port: 3020,
+    serviceUrl: process.env.CONVERSATION_SERVICE
+  },
+  {
+    defaultUrl: 'https://api.ciscospark.com/v1',
+    env: 'HYDRA_SERVICE_URL',
+    name: 'hydra',
+    port: 3030,
+    serviceUrl: process.env.HYDRA_SERVICE_URL
+  },
+  {
+    defaultUrl: 'https://wdm-a.wbx2.com/wdm/api/v1',
+    env: 'WDM_SERVICE_URL',
+    name: 'wdm',
+    port: 3040,
+    serviceUrl: process.env.WDM_SERVICE_URL
+  }
+];
+
+let proxies;
+
+/**
+ * Start yakbak proxy servers for each service
+ * and return an array of those servers.
+ * @returns {Promise}
+ */
+async function startProxies() {
+  await Promise.all(services.map((service) => setEnv(service)));
+  return Promise.all(services.map((service) => start(service)));
+}
+
+/**
+ * Stop each of the proxy servers
+ * in the given array.
+ * @param {Array} proxies
+ * @returns {Promise}
+ */
+async function stopProxies() {
+  if (proxies && proxies.length) {
+    return Promise.all(proxies.map((proxy) => stop(proxy)));
+  }
+  return Promise.resolve();
+}
+
+/**
+ * Sets the process's environment variable given the service,
+ * e.g., HYDRA_SERVICE_URL="http://localhost:3010"
+ * @param {Object} service
```

Comment:
Should we make service a defined jsdoc type so it is easy to look up what props it should have?